src/serverlessFunction/functions.ts

```diff
@@ -364,4 +373,22 @@ export class Functions {
         }
         return null;
     }
+
+    private async isDockerRunning(): Promise<boolean> {
+        return new Promise<boolean>((resolve) => {
+            try {
+                const docker = new Dockerode();
```

Comment:
@vrubezhny and @datho7561 
I just used [dockerode](https://www.npmjs.com/package/dockerode) npm module to check the docker availability. Kindly check it on fedora, I was tested with both windows and Mac and it is working as expected.

Reply:
This works for me, but please explain: how should `func run --build ...` work in case of `podman` and `docker` are installed but `docker` is not running/disabled? (See: https://github.com/redhat-developer/vscode-openshift-tools/pull/3854#issuecomment-1938577036)

Reply:
So, `docker` looks like required to run `func run ...`/`func run --build ...`

Reply:
> So, `docker` looks like required to run `func run ...`/`func run --build ...`

yes It required